byway/src/main/java/com/google/sps/servlets/StopsServlet.java

```diff
@@ -14,19 +14,37 @@
 
 package com.google.sps.servlets;
 
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
-public class DataServlet extends HttpServlet {
+/** Servlet that handles data for stops */
+@WebServlet("/api/stop")
+public final class StopsServlet extends HttpServlet {
+  @GuardedBy("this")
+  private ArrayList<String> stops= new ArrayList<String>();
+  private final Gson gson = new Gson(); 
+  // Implement datastore in routepage branch
 
+  @GuardedBy("this")
```

Comment:
You should make these methods `synchronized`, not mark these methods as `GuardedBy`. You mark the variable as guarded by, then make the methods that access it synchronized.

Reply:
I saw examples where the annotation was used for both variable and method so I just did that. Just changed it to `synchronized`.

Reply:
When you put `@GuardedBy` on a method, you're saying that whoever calls that method needs to hold the specified lock when calling it. However, since `doGet` and `doPost` are servlet methods which override a base class, and the methods on the base class do no have guarded by annotations, there's no way to enforce that these methods are called under a guard.

Reply:
Oh okay, thanks for explaining.